### DISCUSSION TITLE
selected-row in script mode

### DISCUSSION THREAD

**Discussion by numberokta** [davatorium/rofi#1640] [Q&A] [answered]
hi, i want to enable selected-row in script mode by using **rofi_view_set_selected_line** or **listview_set_selected** function. is there any way to get **state** or **listview** variable in _source/modes/script.c_ like **state->list_view** in _source/modes/dmenu.c_ ?
or using other ways in _source/mode.c_ ?
or any other ways ?

**Comment by DaveDavenport** [comment]
No, this won't work and should not be implemented.
It will not work as expected in all use-cases.
There is a reason why there is an abstraction between a mode and the UI.  (dmenu is an exception and because of this also has a lot of limitations).

**Comment by numberokta** [comment, marked as answer]
You're right, this won't work this way. But, I have done it another way.
It's actually very easy to implement and works perfectly.
